private/buf/bufgen/config.go

```diff
@@ -229,11 +235,11 @@ func checkPathAndStrategyUnset(id string, plugin ExternalPluginConfigV1, pluginI
 	return nil
 }
 
-func newManagedConfigV1(externalManagedConfig ExternalManagedConfigV1) (*ManagedConfig, error) {
-	if !externalManagedConfig.Enabled && !externalManagedConfig.IsEmpty() {
-		return nil, errors.New("managed mode options are set but 'managed.enabled: true' is not set")
-	}
+func newManagedConfigV1(logger *zap.Logger, externalManagedConfig ExternalManagedConfigV1) (*ManagedConfig, error) {
 	if !externalManagedConfig.Enabled {
+		if !externalManagedConfig.IsEmpty() && logger != nil {
+			logger.Sugar().Warn("managed mode has options present, but is not enabled")
```

Comment:
nit: we should keep the wording consistent with the previous message. maybe:

```suggestion
			logger.Sugar().Warn("managed mode options are set but are not enabled")
```